### DISCUSSION TITLE
Translation

### DISCUSSION THREAD

**Discussion by rafaelgfirmino** [confetti-framework/confetti#126] [Ideas]
Hi, guys

I found this project very interesting and I would like contribute

So, I can propose to translate using Vuepress and Crowding.  

In the past, I contributed to the [electron i18n](https://github.com/electron/i18n) documentation  which uses [crowdin](https://crowdin.com/project/abp-io/settings#integration-github) to manage translation. We could use it in ABP, it's pretty easy.

Each language would have the same amount of files in derivated English language.
There is no need to create a path file with a new language since Crowdin does this for us. We needed to add a translation in Crowdin panel and it creating a path for us.
Translations are centralized and more people can contribute and fix typos.
if it is interesting I already did a part of the internationalization using Vuepress. it would be just to make a pull request

**Comment by reindert-vetter** [comment]
That sounds interesting! A pull request is welcome.

If a new chapter is added or a part is edited, what causes an update to take place?

**Reply by rafaelgfirmino** [reply]
We can use the github actions to update files in crowdin.
https://support.crowdin.com/github-crowdin-action/

I do a POC to demonstrate in  [Crowdin](https://crowdin.com/project/confettidemo)

[Doc repo for multiple languages](https://github.com/rafaelgfirmino/docs) the URI change in this case, would be like this https://www.confetti-framework.com/docs/en-US/...

I create in this POC a file config.js in an en-us, this file contains a nav and sidebar, the URI is different for each lang. A solution for this processes is to add a Github action to change this URI's

see the POC if it is interesting I do the pull request. Because you need to create the Crowding account and create Github actions

**Reply by reindert-vetter** [reply]
Looks good! Only I cannot register as an Open-source project. That's because:
https://crowdin.com/page/open-source-project-setup-request
`4. You do not have any commercial products related to the open-source project you are requesting a license for.`

If Confetti is a huge success, I will make all kinds of nice tools that I will probably charge money for. Or Confetti will become part of the Laravel community, which also has many "commercial products related to the open-source projects".

What do you think?
